nucypher/network/middleware.py

```diff
@@ -77,6 +77,21 @@ def reencrypt(self, work_order):
             cfrags_and_signatures)
         return cfrags
 
+    def revoke_arrangement(self, ursula, revocation):
+        # TODO: Implement revocation confirmations
+        response = requests.delete("https://{}/kFrag/{}".format(ursula.rest_interface,
+                                                                revocation.arrangement_id.hex()),
+                                                                b''.join(revocation),
+                                                                verify=ursula.certificate_filepath)
+        if response.status_code == 200:
+            return response
+        elif response.status_code == 404:
+            raise RuntimeError("404 - KFrag doesn't exist to revoke with id {}".format(revocation.arrangement_id))
```

Comment:
Why the `RuntimeError` though?

Reply:
Sorta a placeholder, I guess. Not really sure what to give it, but it made sense as a `RuntimeError` at the time.

Reply:
I think it deserves it's own exception subclass :-)

Reply:
And status code inside the exception message?

Reply:
The 404 status code should be included in the Exception, like you did below.